adserver/models.py

```diff
@@ -809,6 +809,19 @@ def state(self):
             return FLIGHT_STATE_UPCOMING
         return FLIGHT_STATE_PAST
 
+    @classmethod
+    def order_flights(cls, flights):
+        """Orders flights by our normal ordering of state, then date, then name."""
+        state_order = {
+            FLIGHT_STATE_CURRENT: 0,
+            FLIGHT_STATE_UPCOMING: 1,
+            FLIGHT_STATE_PAST: 2,
+        }
+        return sorted(
+            flights,
+            key=lambda f: (state_order[f.state], -1 * f.start_date.toordinal(), f.name),
+        )
```

Comment:
Would it be easier to use the `order_by` with multiple arguments? https://docs.djangoproject.com/en/4.0/ref/models/querysets/#django.db.models.query.QuerySet.order_by

In general it's nice to keep querysets as querysets, whereas this is turning them into a list. Probably fine for most use cases, but could be nicer if there isn't a reason not to use `order_by`.

Reply:
Previously, we were using `order_by` with 3 arguments. However, that does not deterministically order the states (which is not a field). We would have to either store the state or change what the state means:

* A flight is currently upcoming if it is live or not as long as the start date is after today. This causes issues with DB ordering because ordering by start date or the live date doesn't definitely do it.
* A flight is current only if it is live AND the start date is before today
* Past is anything else

Reply:
Makes sense. Perhaps worth including this in a comment so we remember why we did this in the future.